planner/core/rule_decorrelate.go

```diff
@@ -189,6 +189,24 @@ func (s *decorrelateSolver) optimize(ctx context.Context, p LogicalPlan, opt *lo
 			}
 			appendRemoveProjTraceStep(apply, proj, opt)
 			return s.optimize(ctx, p, opt)
+		} else if li, ok := innerPlan.(*LogicalLimit); ok {
+			//If offset is not zero, subquery will become always true or always false. This scenario optimizer has internal optimizations.
```

Comment:
This comment should be put at `if li.Offset = 0`.
And there should be a space between the `//` and the first char.